src/components/UserInputComponent/BoatMenuComponent.js

```diff
@@ -31,35 +44,231 @@ export default function BoatMenuComponent(props) {
 
   function handleMenuItemClick(event, newValue) {
     // Calls parent component's (UserInputForm) function with new boat
-    props.setChosenBoatFormikValue(newValue);
+    setChosenBoatFormikValue(newValue);
     setSelectedBoat(newValue);
+    setLength(newValue?.PITUUS || "");
+    setBeam(newValue?.LEVEYS || "");
+    setDraft(newValue?.SYVAYS || "");
     // Ternary operator needed since when the user clears the field, this is run and newValue is null
     setBoatInputString(newValue ? formatInputString(newValue) : "");
   }
 
+  function setChosenBoatFormikValue(newBoat) {
+    if (newBoat) {
+      formik.setValues({
+        ...formik.values,
+        boat: {
+          ...formik.values.boat,
+          length: newBoat.PITUUS || "",
+          beam: newBoat.LEVEYS || "",
+          draft: newBoat.SYVAYS || "",
+        },
+      });
+    } else {
+      formik.setValues({
+        ...formik.values,
+        boat: {
+          ...formik.values.boat,
+          length: "",
+          beam: "",
+          draft: "",
```

Comment:
```suggestion
          length: 210,
          beam: 30,
          draft: 10,
```
These default values could be updated to match the ones found in _src/constants/UserInputDefault_. Since if the user chooses a boat, and then empties the field, then the values are set to empty strings and formik errors are shown.

Reply:
i guess but let's then map the actual default values 😀